include/RMGVertexConfinement.hh

```diff
@@ -91,33 +111,140 @@ class RMGVertexConfinement : public RMGVVertexGenerator {
       return fGeomVolumeData;
     }
 
+    /**
+     * An object which we can generate position samples in. Based on either a
+     * @c G4VPhysicalVolume or geometrical volume defined by a @c G4VSolid . The
+     * sampling can be performed either on the surface or in the volume of the solid.
+     */
     struct SampleableObject {
 
+
         SampleableObject() = default;
         SampleableObject(const SampleableObject&) = default;
+
+        /**
+         * @brief SampleableObject constructor.
+         *
+         * @param v The physical volume.
+         * @param r A rotation matrix for the sampling solid.
+         * @param t A translation vector for the sampling solid.
+         * @param s A solid for geometrical volume sampling or for generating candidate points for
+         * rejection sampling.
+         * @param ns A flag of whether the solid is natively sampeable.
+         * @param ss A flag of whether the solid should be sampled on the surface.
+         */
         SampleableObject(G4VPhysicalVolume* v, G4RotationMatrix r, G4ThreeVector t, G4VSolid* s,
```

Comment:
should we use better argument names? I don't remember why I chose these stupid ones...